### DISCUSSION TITLE
Jinja integration

### DISCUSSION THREAD

**Discussion by thepabloaguilar** [frankie567/starlette-csrf#10] [General]
Hello, thanks for your work on this lib! I'm using it in my FastAPI app and would like to know how I can integration the tokens with my Jinja templates like we do with Django. Is there a way?

Thanks

**Comment by frankie567** [comment]
Hi @thepabloaguilar 👋 

So I guess your goal would be to have the CSRF token in a hidden input field, right? If so, I fear that it won't work very will with this library since it'll check for the challenge in a header, not in the form payload. For this purpose, I suggest you check out [asgi-csrf](https://github.com/simonw/asgi-csrf) which is designed to work this way.

**Reply by thepabloaguilar** [reply]
Yeah, exactly! It seems that `asgi-csrf` is what I was looking for, thanks a lot!

**Reply by abreumatheus** [reply]
With htmx we can pass a header to all requests done inside the body of a template. Any way we can access the token inside a template?

**Reply by frankie567** [reply]
I fear not. You should check out [asgi-csrf](https://github.com/simonw/asgi-csrf) which places the token in the request scope.
